specifyweb/frontend/js_src/lib/components/Molecules/ResourceLink.tsx

```diff
@@ -39,9 +40,12 @@ export function ResourceLink<COMPONENT extends typeof Link['Icon']>({
     >,
     'dialog' | 'onAdd' | 'onClose' | 'onSaved'
   >;
+  readonly autoClose?: boolean;
 }): JSX.Element {
   const [isOpen, _, handleClose, handleToggle] = useBooleanState();
-  React.useEffect(handleClose, [resource]);
+  React.useEffect(() => {
+    if (autoClose === undefined || autoClose) handleClose();
```

Comment:
fyi: `autoClose === undefined || autoClose` is good, and `autoClose !== false` would be equivalent
both are fine
(and both won't be needed once you set the default value)